### PR TITLE
Create Web App with MVC series: Fix major blocking issues

### LINKED ISSUE #20486: 5.0 Intro to MVC cleanup pass
Do another pass and fix problems on the series.

**Scope:**
Blocking issues/major headaches such as but not limited to:
- Screenshots with significant (nearly blocking) mismatches.
- Code highlighting that is showing the wrong thing, 
- Code sample that does not match.

**Outside of the scope of this issue** 
(to be follow up in another tracking issue):
- remove unnecessary includes
- Machine translation improvements
- Style guide issues
- General readability improvement 
 

---
#### Document Details

⚠ *Do not edit this section. It is required for docs.microsoft.com ➟ GitHub issue linking.*

* ID: faf18d20-d28f-1efa-d329-181fff3630fa
* Version Independent ID: 46a7c2e0-4871-491f-02c2-54f7df2237b1
* Content: [Get started with ASP.NET Core MVC](https://docs.microsoft.com/en-us/aspnet/core/tutorials/first-mvc-app/start-mvc?view=aspnetcore-5.0&tabs=visual-studio)
* Content Source: [aspnetcore/tutorials/first-mvc-app/start-mvc.md](https://github.com/dotnet/AspNetCore.Docs/blob/master/aspnetcore/tutorials/first-mvc-app/start-mvc.md)
* Product: **aspnet-core**
* Technology: **aspnetcore-tutorials**
* GitHub Login: @Rick-Anderson
* Microsoft Alias: **riande**

### LINKED ISSUE #20584: Scaffolding fails - Install Package Microsoft.VisualStuido.Web.CodeGeneration.Design required
Scaffolding fails - Install Package Microsoft.VisualStuido.Web.CodeGeneration.Design required.
Should this be included in the tutorial someplace before running scaffolding?

---
#### Document Details

⚠ *Do not edit this section. It is required for docs.microsoft.com ➟ GitHub issue linking.*

* ID: 1d975a83-1550-b284-d0d8-7fe2ea206ff5
* Version Independent ID: 48e81b42-7c2b-afbf-71e6-d4f442a4bd1f
* Content: [Part 4, add a model to an ASP.NET Core MVC app](https://docs.microsoft.com/en-us/aspnet/core/tutorials/first-mvc-app/adding-model?view=aspnetcore-5.0&tabs=visual-studio)
* Content Source: [aspnetcore/tutorials/first-mvc-app/adding-model.md](https://github.com/dotnet/AspNetCore.Docs/blob/master/aspnetcore/tutorials/first-mvc-app/adding-model.md)
* Product: **aspnet-core**
* Technology: **aspnetcore-tutorials**
* GitHub Login: @Rick-Anderson
* Microsoft Alias: **riande**

### PR DESCRIPTION
**Pull request by wadepickett**
[Internal Review](https://review.docs.microsoft.com/en-us/aspnet/core/tutorials/first-mvc-app/start-mvc?view=aspnetcore-5.0&branch=pr-en-us-20530&tabs=visual-studio)

Fixes #20486 
Fixes #20584 

**Scope:
The point is to quickly take care of any version 5 blocking issues/major headaches such as but not limited to:**

- Screenshots with significant (nearly blocking) mismatches.
- Code highlighting that is showing the wrong thing,
- Code sample that does not match.

**Outside of the scope of this issue**
(to be follow up in another tracking issue):

- remove unnecessary includes
- Machine translation improvements
- Style guide issues
- General readability improvement

### CODE REVIEW

**Comment by wadepickett** [comment]
Only up to add-controllers for all three IDE tabs at this point for the first draft commit.

File: aspnetcore/tutorials/first-mvc-app/adding-controller.md [hunk 1]
```diff
@@ -31,19 +31,19 @@ We cover these concepts in this tutorial series and show you how to use them to
 # [Visual Studio](#tab/visual-studio)
 
 * In the **Solution Explorer**, right-click **Controllers > Add > Controller**
-  ![Solution Explorer, right click Controllers > Add > Controller](~/tutorials/first-mvc-app/adding-controller/_static/add_controllerCopy.png)
+  ![Solution Explorer, right click Controllers > Add > Controller](~/tutorials/first-mvc-app/adding-controller/_static/5/add_controllerCopy.png)
```
Comment (Rick-Anderson) [inline aspnetcore/tutorials/first-mvc-app/adding-controller.md:34]: Why move it? Please test with

```suggestion
  ![Solution Explorer, right click Controllers > Add > Controller](~/tutorials/first-mvc-app/adding-controller/_static/add_controller.png)
```

File: aspnetcore/tutorials/first-mvc-app/adding-controller.md [hunk 2]
```diff
@@ -31,19 +31,19 @@ We cover these concepts in this tutorial series and show you how to use them to
...
 
 * In the **Add Scaffold** dialog box, select **MVC Controller - Empty**
 
-  ![Add MVC controller and name it](~/tutorials/first-mvc-app/adding-controller/_static/acCopy.png)
+  ![Add MVC controller and name it](~/tutorials/first-mvc-app/adding-controller/_static/5/acCopy.png)
```
Comment (Rick-Anderson) [inline aspnetcore/tutorials/first-mvc-app/adding-controller.md:38]: Same as previous suggestion. It's a bug in the system that it didn't render the correct image. If the original image doesn't work, delete it, but keep a name change so we don't hit this bug.

File: aspnetcore/tutorials/first-mvc-app/adding-controller.md (hunk 1 above)
Comment (Rick-Anderson) [reply aspnetcore/tutorials/first-mvc-app/adding-controller.md:34]: 5 implies it's different in .NET 5, which it isn't.

**Review by Rick-Anderson — CHANGES_REQUESTED** [review]

File: aspnetcore/tutorials/first-mvc-app/adding-controller.md (hunk 2 above)
Comment (wadepickett) [reply aspnetcore/tutorials/first-mvc-app/adding-controller.md:38]: Thanks Rick!  ( Just a heads up, this is in draft and not ready for review.  I'm still pushing lots of updates to many files.)

You asked why move the screenshots:  I was not moving the existing, those need to stay, I created new version 5 specific screenshots where needed related to the latest VS, while the earlier versions need to be maintained for the VS/SDK in prereqs for those versions.

Comment (Rick-Anderson) [reply aspnetcore/tutorials/first-mvc-app/adding-controller.md:38]: That's not 5 specific, that's VS specific . This block is in moniker 3 and greater. My file was created with the .NET 5 and the current VS.

I realize it's in draft.

Comment (wadepickett) [reply aspnetcore/tutorials/first-mvc-app/adding-controller.md:38]: @Rick-Anderson.  Yes, I understand the screenshot is not 5 specific of course, it is specific to the VS prereq we have on the version of the docs which are versioned by SDK, not IDE.   I saw many examples of screenshots grouped to the doc sdk version and was following that pattern.  Sounds like I shouldn't.

How do you want to specify differences between VS and VC and VS for Mac versions for screenshots?  I can put it in the screenshot name, or I can create new directories for each, but that gets a bit busy.

It looked like there was minor change in VS 16.8.  They were minor, but you can see the difference, and it was a little confusing to see a different one so I updated it. acCopy: Controller on the left menu moved under MVC.  It was not at the top level, for me anyway.  How are we maintaining a VS 2017 for previous docs where that is the prereq?  Are we just paving over?  I saw many examples in the repo where we are maintaining them, so I assumed that was the pattern to follow and it seemed to make sense.

I will also go back and correct for moniker version, thanks.

Comment (Rick-Anderson) [reply aspnetcore/tutorials/first-mvc-app/adding-controller.md:38]: > I will also go back and correct for moniker version, thanks.

No, that image should work for both versions.

> How do you want to specify differences between VS and VC and VS for Mac versions for screenshots? I can put it in the screenshot name

That works.

Comment (wadepickett) [reply aspnetcore/tutorials/first-mvc-app/adding-controller.md:38]: Here is acCopy so the differences are easy to see.   The dialog name and menu are different.
image

Comment (Rick-Anderson) [reply aspnetcore/tutorials/first-mvc-app/adding-controller.md:38]: That one for sure needed an update, the previous, less so.

**Comment by wadepickett** [comment]
Doc builds are stacked and behind per docs portal.  May take an hour our two before one appears.

**Comment by Rick-Anderson** [comment]
They're completely stopped, must be for a system upgrade.  Probaby back up by Monday.

**Comment by wadepickett** [comment]
@Rick-Anderson Thanks!  Working on a new commit right now.  I realize we need this soon.

**Comment by wadepickett** [comment]
I am already moving on general clean up on this series in a new PR while this one for bigger blockers is reviewed and hopefully merges.

**Comment by wadepickett** [comment]
Converted back to draft.  I need to fix the original issue of the Ratings field .cshtml which needs to be updated.

**Review by Rick-Anderson — APPROVED** [review]

**Pull request merged**